pkg/cloud/services/ec2/instances_test.go

```diff
@@ -45,6 +47,69 @@ import (
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+type runInstancesInputMatcher struct {
+	runInstancesInput *ec2.RunInstancesInput
+}
+
+func (m runInstancesInputMatcher) Matches(arg interface{}) bool {
+	riiArg := arg.(*ec2.RunInstancesInput)
```

Comment:
Do we need some kind of error check here in case the arg cannot be cast to the proper type?

Reply:
```suggestion
	riiArg, ok := arg.(*ec2.RunInstancesInput)
	if !ok {
		return false
	}
```

Something like this?

Reply:
Good catch, we do actually need this. I'll apply this patch now.